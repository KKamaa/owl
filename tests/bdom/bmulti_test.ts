import { makeTestFixture, renderToBdom, snapshotTemplateCode } from "../helpers";

let fixture: HTMLElement;

beforeEach(() => {
  fixture = makeTestFixture();
});

describe("BMulti", () => {
  test("removing/adding elements", () => {
    const template = `
        <p>1</p>
        <p t-if="flag">2</p>`;

    snapshotTemplateCode(template);

    const bdom = renderToBdom(template, { flag: false });
    bdom.mount(fixture, [], []);
    expect(fixture.innerHTML).toBe("<p>1</p>");

    const bdom2 = renderToBdom(template, { flag: true });
    bdom.patch(bdom2, [], []);
    expect(fixture.innerHTML).toBe("<p>1</p><p>2</p>");

    const bdom3 = renderToBdom(template, { flag: false });
    bdom.patch(bdom3, [], []);
    expect(fixture.innerHTML).toBe("<p>1</p>");

    const bdom4 = renderToBdom(template, { flag: true });
    bdom.patch(bdom4, [], []);
    expect(fixture.innerHTML).toBe("<p>1</p><p>2</p>");
  });

  test("removing an BMulti with an empty slot", () => {
    const template = `
        <p>1</p>
        <p t-if="flag">2</p>`;

    snapshotTemplateCode(template);

    const bdom = renderToBdom(template, { flag: false });
    bdom.mount(fixture, [], []);
    expect(fixture.innerHTML).toBe("<p>1</p>");

    bdom.remove();
    expect(fixture.innerHTML).toBe("");
    expect(fixture.childNodes.length).toBe(0);
  });
});
