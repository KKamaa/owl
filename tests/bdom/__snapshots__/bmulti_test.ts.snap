// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BMulti removing an BMulti with an empty slot 1`] = `
"function anonymous(Blocks, utils
) {
  let {BCollection, BComponent, BComponentH, BHtml, BMulti, BNode, BStatic, BText, BDispatch} = Blocks;
  let {elem, toString, withDefault, call, getTemplate, zero, callSlot, capture} = utils;
  
  class Block1 extends BStatic {
    static el = elem(\`<p>1</p>\`);
  }
  
  class Block2 extends BStatic {
    static el = elem(\`<p>2</p>\`);
  }
  
  return function template(ctx, node = ctx.__owl__) {
    const b1 = new BMulti(2);
    const b2 = b1.children[0] = new Block1();
    if (ctx['flag']) {
      const b3 = b1.children[1] = new Block2();
    }
    return b1;
  }
}"
`;

exports[`BMulti removing/adding elements 1`] = `
"function anonymous(Blocks, utils
) {
  let {BCollection, BComponent, BComponentH, BHtml, BMulti, BNode, BStatic, BText, BDispatch} = Blocks;
  let {elem, toString, withDefault, call, getTemplate, zero, callSlot, capture} = utils;
  
  class Block1 extends BStatic {
    static el = elem(\`<p>1</p>\`);
  }
  
  class Block2 extends BStatic {
    static el = elem(\`<p>2</p>\`);
  }
  
  return function template(ctx, node = ctx.__owl__) {
    const b1 = new BMulti(2);
    const b2 = b1.children[0] = new Block1();
    if (ctx['flag']) {
      const b3 = b1.children[1] = new Block2();
    }
    return b1;
  }
}"
`;
